core/src/advice/map.rs

```diff
@@ -17,32 +21,47 @@ use crate::utils::{ByteReader, ByteWriter, Deserializable, DeserializationError,
 /// associated with a given key onto the advice stack using `adv.push_mapval` instruction. The VM
 /// can also insert new values into the advice map during execution.
 #[derive(Debug, Clone, Default, PartialEq, Eq)]
-pub struct AdviceMap(BTreeMap<Word, Vec<Felt>>);
+pub struct AdviceMap(BTreeMap<Word, Arc<[Felt]>>);
 
 /// Pair representing a key-value entry in an [`AdviceMap`]
-type MapEntry = (Word, Vec<Felt>);
+type MapEntry = (Word, Arc<[Felt]>);
 
 impl AdviceMap {
     /// Creates a new advice map.
     pub fn new() -> Self {
```

Comment:
Should we remove?

Reply:
I think clippy complains if you have a `Default` implementation with a `new()`